tests/op_coreutils/bigquery/test_write.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+import pytest
+from unittest.mock import patch, MagicMock
+from datetime import date
+import polars as pl
+import pandas as pd
+import numpy as np
+from op_coreutils.bigquery.write import (
+    overwrite_table,
+    overwrite_partition_static,
+    overwrite_partitions_dynamic,
+    upsert_partition,
+    OPLabsBigQueryError,
+)
+
+# Generate a larger sample DataFrame for testing
+date_range = pd.date_range("2024-01-01", periods=100, freq="D")
+num_rows = 1000
+
+# Create random data for each column
+test_df = pl.DataFrame(
+    {
+        "dt": np.random.choice(date_range, num_rows),
+        "data": np.random.randint(100, 500, size=num_rows),
+        "category": np.random.choice(["A", "B", "C", "D"], num_rows),
+        "value": np.random.uniform(1.0, 10.0, size=num_rows).round(2),
+    }
+)
+
+# Sort by 'dt' for consistency
+test_df = test_df.sort("dt")
+
+
+@patch("op_coreutils.bigquery.write.init_client")
```

Comment:
We should not need to patch the init_client method. The method itself should be "env aware", so if we are running in tests you will get a MagicMock withou having to patch it in each test yourself.

Reply:
I was pulling the init_client from `packages/op-coreutils/src/op_coreutils/gsheets.py` instead of `packages/op-coreutils/src/op_coreutils/bigquery/write.py`. 
I corrected this throughout the tests and removed the redundant patches.